vernac/assumptions.ml

```diff
@@ -190,14 +190,27 @@ let fold_with_full_binders g f n acc c =
       Array.fold_left (fun acc (t,b) -> f n' (f n acc t) b) acc fd
   | Array(_u,t,def,ty) -> f n (f n (Array.fold_left (f n) acc t) def) ty
 
+let get_constant_body kn =
+  let cb = lookup_constant kn in
+  let env = Global.env () in
+  let access = Library.indirect_accessor in
+  let otab = Environ.opaque_tables env in
+  match cb.const_body with
+  | Undef _ | Primitive _ -> None
+  | Def c -> Some (Mod_subst.force_constr c)
+  | OpaqueDef o ->
+    match Opaqueproof.force_proof access otab o with
+    | c, _ -> Some c
+    | exception _ -> None (* missing delayed body, e.g. in vok mode *)
```

Comment:
Should we really catch everything?
Maybe force_proof should return an option or have a dedicated exception.

Reply:
This should be done eventually but since I am planning to mess with this code, I prefer submitting this PR as a fix first.